src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -423,6 +423,11 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         }
     }
 
+    hstring ProfileViewModel::TabDisplayName()
+    {
+        return TabTitle().empty() ? Name() : TabTitle();
```

Comment:
when the Name or TabTitle fields change (make sure you listen to PropertyChanged notifications), you need to tell xaml that TabDisplayName changed

Reply:
Ah of course, thanks for catching that!